cpp/include/cudf/strings/combine.hpp

```diff
@@ -158,6 +118,143 @@ std::unique_ptr<column> concatenate(
+ *        Default is an empty string.
+ * @param narep String that should be used in place of any null strings
+ *        found in any column. Default of invalid-scalar means any null entry in any column will
+ *        produces a null result for that row.
+ * @param mr Device memory resource used to allocate the returned column's device memory.
+ * @return New column with concatenated results.
+ */
+std::unique_ptr<column> concatenate(
+  table_view const& strings_columns,
+  string_scalar const& separator      = string_scalar(""),
+  string_scalar const& narep          = string_scalar("", false),
+  rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
+
+/**
+ * @brief Given a lists column of strings (each row is a list of strings), concatenates the strings
+ * within each row and returns a single strings column result.
+ *
+ * Each new string is created by concatenating the strings from the same row (same list element)
+ * delimited by the row separator provided in the `separators` strings column.
+ *
+ * A null list row will always result in a null string in the output row. Any non-null list row
+ * having a null entry will result in the corresponding output row to be null unless a valid
+ * `string_narep` scalar is provided to be used in its place. Any null row in the `separators`
+ * column will also result in a null output row unless a valid `separator_narep` scalar is provided
+ * to be used in place of the null separators.
+ *
+ * @code{.pseudo}
+ * Example:
+ * s = [ {'aa', 'bb', 'cc'}, null, {'', 'dd'}, {'ee', null}, {'ff', 'gg'} ]
+ * sep  = ['::', '%%',  '!',  '*',  null]
+ *
+ * r1 = concatenate(s, sep)
+ * r1 is ['aa::bb::cc', null, '!dd', null, null]
+ *
+ * r2 = concatenate(s, sep, ':', '_')
+ * r2 is ['aa::bb::cc', null,  '!dd', 'ee*_', 'ff:gg']
+ * @endcode
+ *
+ * @throw cudf::logic_error if input column is not lists of strings column.
+ * @throw cudf::logic_error if the number of rows from `separators` and `lists_strings_column` do
+ * not match
+ *
+ * @param lists_strings_column   Column containing lists of strings to concatenate
+ * @param separators             Strings column that provides separators for concatenation
+ * @param separator_narep        String that should be used to replace null separator, default is an
+ * invalid-scalar denoting that rows containing null separator will result in null string in the
+ * corresponding output rows
+ * @param string_narep           String that should be used to replace null strings in any
+ * non-null list row, default is an invalid-scalar denoting that list rows containing null strings
+ * will result in null string in the corresponding output rows
+ * @param mr                     Device memory resource used to allocate the returned column's
+ * device memory
+ * @return                       New strings column with concatenated results
+ */
+std::unique_ptr<column> concatenate_rows(
```

Comment:
This isn't really concatenating rows. It's concatenating the elements within a list. I'd go for `concatenate_list_elements`.

Reply:
I like that. And if you have even a better name (shorter) then it's perfect 😄